### PR TITLE
ci(): upgrade to electron@19 + add e2e tests

### PR DESCRIPTION
**Pull request by ShaMan123**
- upgrade to electron@19
   -  upgrade nan binding  
   - couldn't upgrade to higher versions due to the need for context isolation/bridging that requires a major rewrite
- add e2e tests 
   - playwright not working on linux (probably due to old electron version)
   - standalone test on built app is working

### CODE REVIEW

**Comment by abebeos** [comment]
@ShaMan123 , thanks for this. Looks like high-effort to review. Would you be willing to make a separate PR that includes only(!) the minimum changes for an upgrade to electron 19? You should not introduce or alter anything not needed for the upgrade.

**Comment by ShaMan123** [comment]
I can remove the tests. The rest is needed and is the bare minimum.
I am working now on isolating the logic so it can be used as a node module... the code is not in a good condition.
So I am not sure. Maybe it is best to close this PR and try to make it a node module. Then I can bind the electron app with the latest version as a consumer.

**Comment by ShaMan123** [comment]
Moving to electron latest will require this anyways

**Comment by abebeos** [comment]
> this

You mean the decoupling to a node_module?

**Comment by abebeos** [comment]
> I can remove the tests. The rest is needed and is the bare minimum.

I would prefer to have the tests in a follow-up PR.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -12,32 +12,64 @@ jobs:
   build:
     strategy:
       matrix:
-        node-version: [14.x, 16.x, 18.x, 20.x]
-        os: [windows-2022, ubuntu-22.04]
+        # os: [windows-2022, ubuntu-latest]
```
Comment (abebeos) [inline .github/workflows/build.yml:15]: avoid the use of "ubuntu-latest", future updates can break the build. Better to pin on the versions you know to work.

**Comment by ShaMan123** [comment]
> > this
> 
> You mean the decoupling to a node_module?

yes

File: .github/workflows/build.yml (hunk 1 above)
Comment (ShaMan123) [reply .github/workflows/build.yml:15]: np

**Comment by ShaMan123** [comment]
> > I can remove the tests. The rest is needed and is the bare minimum.
> 
> I would prefer to have the tests in a follow-up PR.

What do you think in terms of strategy? Shall we go for the node module decoupling or first we merge this since it is less disruptive and is based on this head anyways?

**Comment by abebeos** [comment]
> What do you think in terms of strategy?

Me dizzy. First day I look again at code after a longer general break. If you feel confident that you can go through with this, then choose the strategy yourself.

Just try to have easy verifiable chunks (PRs). I'll give you access-rights, to fasten things up.

**Comment by abebeos** [comment]
> strategy

- PR update to electron 19
- PR add more tests
- PR(s) node_module stuff for further electron updates, keep it in increments, avoid super-PRs.

**Comment by abebeos** [comment]
And a general remark: you need to avoid single-commit PRs. Leav the commits for easier review, you can then later squash down the commit-count, sometimes even to one.

**Comment by ShaMan123** [comment]
Will do. I will close this PR and start a new one.
At some point I suggest introducing prettier and lint for code quality and doing a whitespace PR (very disruptive, non reviewable)

**Comment by abebeos** [comment]
> At some point I suggest introducing prettier and lint for code quality

why not.

You should have gotten some kind of invite here on github.

**Comment by ShaMan123** [comment]
Got it, thanks.
I am removing `electron-settings` - seems to cause issues.
Any problems? Should I persist the settings in some app folder?

**Comment by ShaMan123** [comment]
Or shall I use localStorage? Is that valid with electron? Does it persist after a new version?

**Comment by ShaMan123** [comment]
closing in favor of #159

**Pull request closed without merging**